kratos/geometries/geometry.h

```diff
@@ -554,26 +559,44 @@ class Geometry
             *i = typename PointType::Pointer( new PointType( **i ) );
     }
 
-     // virtual Kratos::shared_ptr< Geometry< Point > > Clone() const
-     // {
-     //     Geometry< Point >::PointsArrayType NewPoints;
+    ///@}
+    ///@name Id
+    ///@{
 
-     //     //making a copy of the nodes TO POINTS (not Nodes!!!)
+    /// Id of this Geometry
+    IndexType Id() const
+    {
+        return mId;
+    }
+
+    /// Sets Id of this Geometry
+    void SetId(IndexType Id)
+    {
+        KRATOS_ERROR_IF(IsSelfAssignedId(Id) && IsGeometryIdString(Id))
+            << "Id out of range. The first bit of the Id is used to "
+            << "detect if Id is int or hash of name. Second bit defines"
+            << "if Id is self assigned or not."
+            << std::endl;
 
-     //     for ( IndexType i = 0 ; i < this->size() ; i++ )
-     //     {
-     //        NewPoints.push_back(Kratos::make_shared< Point >((*mpPointVector)[i]));
-     //     }
+        mId = Id;
+    }
 
-     //     //NewPoints[i] = typename Point::Pointer(new Point(*mPoints[i]));
+    /// Sets Id with the use of the name of this geometry
+    void SetId(std::string name)
+    {
+        mId = GetNameHash(name);
+    }
 
-     //     //creating a geometry with the new points
-     //     Geometry< Point >::Pointer p_clone( new Geometry< Point >( NewPoints ) );
+    /// Gets the corresponding hash-Id to a string name
+    inline IndexType GetNameHash(std::string name) const
```

Comment:
I would change this to `GenerateId` or `CreateId`

Reply:
I prefer GenerateId because I think it defines better that there is always one corresponding Id to given name.